From: =?utf-8?q?Bj=C3=B6rn_Daase?= <bjoern@daase.net>
Date: Tue, 15 Feb 2022 11:21:48 +0100
Subject: dateMenu: Don't manipulate passed events

The event passed to formatEventTime() is reused at a later point.
Therefore, we are not allowed to manipulate the event directly.
This fixes an issue where the user clicks on a multi-day all-day event
the second time before the event gets garbage collected and the event
then is one day shorter.

Fixes 528ee01fef6e5ca2303fc766be29a854d8ee928b

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2184>

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/0fded45c76
---
 js/ui/dateMenu.js | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/js/ui/dateMenu.js b/js/ui/dateMenu.js
index 310a6bc..75b446f 100644
--- a/js/ui/dateMenu.js
+++ b/js/ui/dateMenu.js
@@ -175,7 +175,7 @@ class EventsSection extends St.Button {
 
     _formatEventTime(event) {
         const eventStart = event.date;
-        const eventEnd = event.end;
+        let eventEnd = event.end;
 
         const allDay =
             eventStart.getTime() === this._startDate.getTime() && eventEnd.getTime() === this._endDate.getTime();
@@ -203,8 +203,10 @@ class EventsSection extends St.Button {
 
             const startYear = eventStart.getFullYear();
 
-            if (endsAtMidnight)
+            if (endsAtMidnight) {
+                eventEnd = new Date(eventEnd);
                 eventEnd.setDate(eventEnd.getDate() - 1);
+            }
 
             const endYear = eventEnd.getFullYear();
 
